### ISSUE TITLE
Inline image size does not make sense

### ISSUE THREAD

**Issue by julian-weinert** [WordPress/gutenberg#28842]
I added an inline image to my article. I specified an arbitrary size and the resulting HTML is pretty much useless.
It looks like this

```
<img width="426" height="650" style="width:150px" sizes="(max-width: 426px) 100vw, 426px">
```

How is this supposed to work? Am I supposed to copy the image width from the library into the width field?

**Comment by talldan** [comment]
I tried testing this with a few different images and sizes, but couldn't reproduce a situation where I had a stretched image.

It does seem possible to delete the width and hit enter, though this adds an invalid style to the image tag (which looks like a bug).

**Comment by julian-weinert** [comment]
Yeah, I tried that. In one instance I got a NaN value and in a different case I wasn't able to delete the width.
It seems to me like a race condition between the form post and the JS validator.
But nevertheless, since it is more likely a bug it's not the best solution ;)

What HTML did you get? I mean it specifies width and height as attributes but only width in the style. Which means that if the width from the block (in the style attribute) is wrong the image gets stretched.

**Comment by chyvak1831** [comment]
**The same thing.**

- Wordpress: 5.7
- Theme: twentytwentyone 1.2

No plugins, no custom code. As I see - for set width inline image is used inline `style width`, but in attribute `sizes` sets largest image width.

As for me - it's definitely bug.
![image](https://user-images.githubusercontent.com/11210100/112894577-b3e4ed80-90e4-11eb-9d8d-9f56bd1a94b5.png)

**Comment by chyvak1831** [comment]
one more clarification: inline CSS image width works correct, **but why it does not affects on sizes**? Why always largest size?
Loads so large image size - no make sense what visually size setup...

![image](https://user-images.githubusercontent.com/11210100/112894863-15a55780-90e5-11eb-8596-480afac11f34.png)

**Comment by getsource** [comment]
Related: #31515

**Comment by julian-weinert** [comment]
Can I please bump this again?
Where in the world does this make sense?
Simply try adding a photo in one "block column". It simply doesn't work...

This thing needs serious rethinking and rewriting. There's just so much wrong, but Wordpress completely pushes this thing. I tried using Gutenberg in my latest project again. It's just pain and grief.

Next website I do with Wordpress will replace Gutenberg with a custom editor and only use Wordpress as page manager and database interface. This is too annoying and has cost me too much time and honestly money...
